packages/ragbits-core/src/ragbits/core/prompt/prompt.py

```diff
@@ -236,7 +236,7 @@ def parse_response(self, response: str) -> OutputT:
         Raises:
             ResponseParsingError: If the response cannot be parsed.
         """
-        return self.response_parser(response)
+        return self.response_parser(response)  # type: ignore
```

Comment:
I don't believe that sprinkling `type: ignore`s is usually the answer. We designed all those classes ourself. They should work properly with each other. If they don't after this changes, they should be redesigned to do (although it's still not clear to me what parsing a stream means).

Reply:
resolved by implementing separate method for streaming